internal/pkg/fleetapi/action.go

```diff
@@ -641,101 +611,13 @@ func (a *Actions) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-// UnmarshalYAML attempts to decode yaml actions.
-func (a *Actions) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	var nodes []FleetAction
-	if err := unmarshal(&nodes); err != nil {
-		return errors.New(err,
-			"fail to decode action",
-			errors.TypeConfig)
-	}
-	actions := make([]Action, 0, len(nodes))
-	for i := range nodes {
-		var action Action
-		n := nodes[i]
-		switch n.ActionType {
-		case ActionTypePolicyChange:
-			action = &ActionPolicyChange{
-				ActionID:   n.ActionID,
-				ActionType: n.ActionType,
-			}
-			if err := yaml.Unmarshal(n.Data, action); err != nil {
-				return errors.New(err,
-					"fail to decode POLICY_CHANGE action",
-					errors.TypeConfig)
-			}
-		case ActionTypePolicyReassign:
-			action = &ActionPolicyReassign{
-				ActionID:   n.ActionID,
-				ActionType: n.ActionType,
-			}
-		case ActionTypeInputAction:
-			action = &ActionApp{
-				ActionID:   n.ActionID,
-				ActionType: n.ActionType,
-				InputType:  n.InputType,
-				Timeout:    n.Timeout,
-				Data:       n.Data,
-				Signed:     n.Signed,
-			}
-		case ActionTypeUnenroll:
-			action = &ActionUnenroll{
-				ActionID:   n.ActionID,
-				ActionType: n.ActionType,
-				Signed:     n.Signed,
-			}
-		case ActionTypeUpgrade:
-			action = &ActionUpgrade{
-				ActionID:         n.ActionID,
-				ActionType:       n.ActionType,
-				ActionStartTime:  n.ActionStartTime,
-				ActionExpiration: n.ActionExpiration,
-				Retry:            n.Retry,
-			}
-			if err := yaml.Unmarshal(n.Data, &action); err != nil {
-				return errors.New(err,
-					"fail to decode UPGRADE_ACTION action",
-					errors.TypeConfig)
-			}
-		case ActionTypeSettings:
-			action = &ActionSettings{
-				ActionID:   n.ActionID,
-				ActionType: n.ActionType,
-			}
-			if err := yaml.Unmarshal(n.Data, action); err != nil {
-				return errors.New(err,
-					"fail to decode SETTINGS_ACTION action",
-					errors.TypeConfig)
-			}
-		case ActionTypeCancel:
-			action = &ActionCancel{
-				ActionID:   n.ActionID,
-				ActionType: n.ActionType,
-			}
-			if err := yaml.Unmarshal(n.Data, action); err != nil {
-				return errors.New(err,
-					"fail to decode CANCEL_ACTION action",
-					errors.TypeConfig)
-			}
-		case ActionTypeDiagnostics:
-			action = &ActionDiagnostics{
-				ActionID:   n.ActionID,
-				ActionType: n.ActionType,
-			}
-			if err := yaml.Unmarshal(n.Data, action); err != nil {
-				return errors.New(err,
-					"fail to decode REQUEST_DIAGNOSTICS_ACTION action",
-					errors.TypeConfig)
-			}
-		default:
-			action = &ActionUnknown{
-				ActionID:     n.ActionID,
-				ActionType:   ActionTypeUnknown,
-				originalType: n.ActionType,
-			}
-		}
-		actions = append(actions, action)
-	}
-	*a = actions
-	return nil
+// UnmarshalYAML prevents to decode actions from .
```

Comment:
```suggestion
// UnmarshalYAML prevents to decode actions from .
```

To decode actions from what?

Reply:
fixed